core/variant/type_info.h

```diff
@@ -367,4 +367,82 @@ ZERO_INITIALIZER_NUMBER(char32_t)
 ZERO_INITIALIZER_NUMBER(float)
 ZERO_INITIALIZER_NUMBER(double)
 
+template <typename T>
+inline constexpr Variant::Type get_variant_type_v = TypeInherits<Object, T>::value ? Variant::OBJECT : Variant::VARIANT_MAX;
```

Comment:
The name here is a bit misleading now isn't it? It's not actually getting the variant type right? It  only gets the variant type for types that inherit `Object`, otherwise it just returns the "dummy" value `Variant::VARIANT_MAX`. Is that right?

Reply:
Ah, I understand now. C++ templates are confusing >.<. This top line is essentially stating that the "default" value, because any specific template instantiation can override this, is that right? In that case, ignore my previous comment.

Also, are we sure this kind of thing wouldn't be more appropriate in `variant.h` or something?

Reply:
I can hardly blame you; it took a good while to wrap my head around templates, and they *still* trip me up from time to time.

I'm inclined to say that this is still best suited in this file; templates need pretty much all of their logic defined ahead of time. I suppose some object-specific defines could happen elsewhere, but this layout keeps everything in a single script.